redshift_sqlalchemy/dialect.py

```diff
@@ -145,19 +187,91 @@ class RedshiftDialect(PGDialect_psycopg2):
+        insp = inspect(connection)
+        key = (schema or insp.default_schema_name, table_name)
+        tables, views = self._get_all_table_and_view_info(connection)
+        table_info = tables[key]
+        column_info = self._get_all_column_info(connection)
+        columns = column_info[key]
+        # If sortkey is interleaved, column numbers alternate negative values
+        keyfunc = lambda col: abs(int(col.sortkey))
+        sortkey_cols = sorted([col for col in columns if col.sortkey],
+                              key=keyfunc)
+        interleaved = any([int(col.sortkey) < 0 for col in sortkey_cols])
+        sortkey = [col.name for col in sortkey_cols]
+        sortkey_type = INTERLEAVED_SORTKEY if interleaved else COMPOUND_SORTKEY
+        distkeys = [col.name for col in columns if col.distkey]
+        distkey = distkeys[0] if distkeys else None
+        return {
+            'redshift_diststyle': table_info.diststyle,
+            'redshift_distkey': distkey,
+            'redshift_sortkey': sortkey,
+            'redshift_sortkey_type': sortkey_type,
+        }
+
+    @reflection.cache
+    def get_columns(self, connection, table_name, schema=None, **kw):
+        all_columns = self._get_all_column_info(connection)
+        insp = inspect(connection)
+        key = (schema or insp.default_schema_name, table_name)
+        these_columns = all_columns[key]
+        domains = self._load_domains(connection)
+        columns = [self._get_column_info(col, domains) for col in these_columns]
+        return columns
+
+    @reflection.cache
     def get_pk_constraint(self, connection, table_name, schema=None, **kw):
-        """
-        Constraints in redshift are informational only. This allows reflection
-        to work.
-        """
-        return {'constrained_columns': [], 'name': ''}
+        all_constraints = self._get_all_constraint_info(connection)
+        insp = inspect(connection)
+        key = (schema or insp.default_schema_name, table_name)
+        constraints = all_constraints.get(key, [])
+        pk_constraints = [c for c in constraints if c.contype == 'p']
+        if not pk_constraints:
+            return {'constrained_columns': [], 'name': ''}
+        pk_constraint = pk_constraints[0]
+        # The slice here removes the 'PRIMARY KEY(' prefix and the ')' suffix
+        column_text = pk_constraint.condef[13:-1]
+        constrained_columns = column_text.split(', ')
+        return {
+            'constrained_columns': constrained_columns,
+            'name': None,
+        }
+
+    @reflection.cache
+    def get_foreign_keys(self, connection, table_name, schema=None, **kw):
+        all_constraints = self._get_all_constraint_info(connection)
+        insp = inspect(connection)
+        key = (schema or insp.default_schema_name, table_name)
+        constraints = all_constraints.get(key, [])
+        fk_constraints = [c for c in constraints if c.contype == 'f']
```

Comment:
Are you talking about the `c` within the list comprehension? Comprehensions are one of the places that single-character names make sense since the scope is so constrained.